toolkit/tools/scheduler/schedulerutils/buildlist.go

```diff
@@ -9,14 +9,61 @@ import (
 	"os"
 	"path/filepath"
 
+	"github.com/juliangruber/go-intersect"
 	"github.com/microsoft/CBL-Mariner/toolkit/tools/imagegen/configuration"
 	"github.com/microsoft/CBL-Mariner/toolkit/tools/imagegen/installutils"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/exe"
 	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/logger"
 	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/pkggraph"
 	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/pkgjson"
 	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/sliceutils"
 )
 
+func ParseAndGeneratePackageList(dependencyGraph *pkggraph.PkgGraph, pkgsToBuild, pkgsToRebuild, pkgsToIgnore, imageConfig, baseDirPath string) (finalPackagesToBuild, packagesToRebuild, packagesToIgnore []*pkgjson.PackageVer, err error) {
+	// Generate the list of packages that need to be built.
+	// If none are requested then all packages will be built.
+	packagesToBuild, err := PackageNamesToBuiltPackages(exe.ParseListArgument(pkgsToBuild), dependencyGraph)
+	if err != nil {
+		err = fmt.Errorf("unable to find build nodes for the packages to build, error:\n%s", err)
+		return nil, nil, nil, err
```

Comment:
>return nil, nil, nil, err [](http://example.com/codeflow?start=2&length=25)

No strong feelings about this, but I think it's fine to drop `nil. nil. nil, err` if we're returning an error. The contents of these lists can be safely regarded "undefined behaviour" and it's really the callers fault, if they try to use an output from a function, which failed to run.

Reply:
Sure, cleaned these up